test/graph/graph-mary-manipulation-test.js

```diff
@@ -39,62 +46,60 @@ describe('Graph manipulation for `mary` stream', function() {
         });
         it('node without an id will get a generated id', function() {
             node.setType('TOK');
-            graph.addNode(node);
             // IDs should increment consecutively.
             assert.equal(node.getId(), (parseInt(graph._generateNodeId()) - 1).toString());
         });
         it('node will be added to the graph', function() {
             var preAddTotalCount = graph.getNodes().length;
             var preAddTypeCount = graph.getNodesByType('TOK').length;
-            node.setType('TOK');
+            node = new Node('900000', graph, 'TOK');
             graph.addNode(node);
             assert.equal(node, graph.getNodeById(node.getId()));
             assert.equal(preAddTotalCount, graph.getNodes().length - 1);
             assert.equal(preAddTypeCount, graph.getNodesByType(node.getType()).length - 1);
         });
         describe('sequence trait', function() {
+            beforeEach(createGraph);
+            beforeEach(function() { node = graph.createNode('TOK', '9000000'); });
+
             it('adding a first node', function() {
-                node.setType('TOK');
-                node.addEdge('24', 'TOK', 'next');
-                graph.addNode(node);
+                node.addEdge('24', 'next');
                 assert.equal(node.next(), graph.getNodeById('24'));
+                assert.equal(node.previous(), undefined);
             });
             it('adding a middle node', function() {
-                node.setType('TOK');
-                node.addEdge('83', 'TOK', 'next');
-                graph.addNode(node);
+                // console.log('24s next edge', graph.getNodeById('24').next()._id);
+                // console.log('83s prev edge', graph.getNodeById('83').previous()._id);
+                node.addEdge('83', 'next');
+                // console.log('Nodes next edge should be 83', node.next()._id);
                 assert.equal(node.previous(), graph.getNodeById('24'));
                 assert.equal(node.next(), graph.getNodeById('83'));
             });
-            it('adding a last node', function() {
```

Comment:
The only test I could think to add, since we currently don't have any logic around last nodes, is one that checks if we link the current last node in a sequence to the new node, the library doesn't try to add a next edge on the new node to anything.